From 6e408752e4e53dafca0eedc1d44f59ef9ab283ee Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stephan=20D=C3=BCnner?= <stephan.duenner@bytesatwork.ch>
Date: Fri, 9 Aug 2019 19:17:33 +0200
Subject: [PATCH] cmd: add cpuinfo

Add command for print_cpuinfo.

Signed-off-by: Rico Ganahl <rico.ganahl@bytesatwork.ch>
(cherry picked from commit 5fc220f6e6835d99bab7c4cef665dca678807bed)
---
 cmd/Kconfig   |  6 ++++++
 cmd/Makefile  |  1 +
 cmd/cpuinfo.c | 19 +++++++++++++++++++
 3 files changed, 26 insertions(+)
 create mode 100644 cmd/cpuinfo.c

diff --git a/cmd/Kconfig b/cmd/Kconfig
index 8a44461972..f17586176e 100644
--- a/cmd/Kconfig
+++ b/cmd/Kconfig
@@ -222,6 +222,12 @@ config CMD_CPU
 	  internal name) and clock frequency. Other information may be
 	  available depending on the CPU driver.
 
+config CMD_CPUINFO
+	bool "cpuinfo"
+	help
+	  Print information about the CPU.
+
+
 config CMD_LICENSE
 	bool "license"
 	select BUILD_BIN2C
diff --git a/cmd/Makefile b/cmd/Makefile
index 3ac7104546..200a7c1b09 100644
--- a/cmd/Makefile
+++ b/cmd/Makefile
@@ -39,6 +39,7 @@ obj-$(CONFIG_CMD_CONFIG) += config.o
 obj-$(CONFIG_CMD_CONITRACE) += conitrace.o
 obj-$(CONFIG_CMD_CONSOLE) += console.o
 obj-$(CONFIG_CMD_CPU) += cpu.o
+obj-$(CONFIG_CMD_CPUINFO) += cpuinfo.o
 obj-$(CONFIG_DATAFLASH_MMC_SELECT) += dataflash_mmc_mux.o
 obj-$(CONFIG_CMD_DATE) += date.o
 obj-$(CONFIG_CMD_DEMO) += demo.o
diff --git a/cmd/cpuinfo.c b/cmd/cpuinfo.c
new file mode 100644
index 0000000000..0882fde1a6
--- /dev/null
+++ b/cmd/cpuinfo.c
@@ -0,0 +1,19 @@
+/*
+ * Copyright (C) 2018 bytes at work AG
+ *
+ * SPDX-License-Identifier:	GPL-2.0+
+ */
+
+#include <common.h>
+#include <command.h>
+
+static int do_cpuinfo(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
+{
+	return print_cpuinfo();
+}
+
+U_BOOT_CMD(
+	cpuinfo, 1, 1, do_cpuinfo,
+	"show cpu info",
+	""
+);
-- 
2.30.2

